### PR TITLE
fix: regex match condition for primary type

### PR DESCRIPTION
**Pull request by jpuri**
This PR is in continuation of the previous one: https://github.com/MetaMask/eth-json-rpc-middleware/pull/350

It fixes a condition while validating primary type of types signatures.

### CODE REVIEW

File: src/utils/common.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * @returns Parameter string with array brackets [] removed.
  */
 export const stripArrayTypeIfPresent = (typeString: string) => {
-  if (typeString?.match(/\S\[\d*\]$/u) !== null) {
+  if (typeString?.match(/\S\[\d*\]$/u)) {
```
Comment (jpuri) [inline src/utils/common.ts:8]: I found this condition not working as expected when `typeString` was `undefined`.

**Review by OGPoyraz — DISMISSED** [review]

File: src/utils/common.ts (hunk 1 above)
Comment (Gudahtt) [reply src/utils/common.ts:8]: Makes sense! Could you add a test case for this?

Comment (jpuri) [reply src/utils/common.ts:8]: PR is updated with test case.

File: src/utils/common.test.ts [hunk 2]
```diff
@@ -10,6 +10,9 @@ describe('CommonUtils', () => {
     it('return types which are not array without any change', () => {
       expect(stripArrayTypeIfPresent('string')).toBe('string');
       expect(stripArrayTypeIfPresent('string []')).toBe('string []');
+      expect(
+        stripArrayTypeIfPresent(undefined as unknown as string),
```
Comment (Gudahtt) [inline src/utils/common.test.ts:14]: Nit: Better to use `ts-expect-error` here instead, so it's clear this is invalid type is intentional. e.g.

```suggestion
        // @ts-expect-error Intentionally testing invalid input
        stripArrayTypeIfPresent(undefined),
```

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: src/utils/common.test.ts (hunk 2 above)
Comment (Gudahtt) [reply src/utils/common.test.ts]: Explained further here: https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#use-ts-expect-error-to-force-runtime-execution-of-a-branch-for-validation-or-testing

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**